### PR TITLE
Add function to transfer balance to address

### PR DESCRIPTION
**Pull request by maikelordaz**

### CODE REVIEW

File: contracts/facets/CollateralFacet.sol [hunk 1]
```diff
@@ -27,6 +27,7 @@ contract CollateralFacet is ICollateral {
     event OnCollateralWithdrawal(
         uint indexed termId,
         address indexed user,
+        address receiver,
         uint indexed collateralAmount
     );
     event OnReimbursementWithdrawn(uint indexed termId, address indexed user, uint indexed amount);
```
Comment (MohammedTaka) [inline contracts/facets/CollateralFacet.sol]: Should this not also have a receiver now?

File: contracts/facets/YGFacetZaynFi.sol [hunk 2]
```diff
@@ -22,14 +27,7 @@ contract YGFacetZaynFi is IYGFacetZaynFi {
...
     function claimAvailableYield(uint termId) external {
-        _claimAvailableYield(termId, msg.sender);
-    }
-
-    /// @notice This function allows a user to claim the current available yield
-    /// @param termId The term id for which the yield is being claimed
-    /// @param user The user address that is claiming the yield
-    function claimAvailableYield(uint termId, address user) external {
-        _claimAvailableYield(termId, user);
+        LibYieldGeneration._claimAvailableYield(termId, msg.sender, msg.sender);
```
Comment (MohammedTaka) [inline contracts/facets/YGFacetZaynFi.sol]: Should there be a receiver input in the event if it's always msg.sender?

File: test/unit/02-combined-tests/combined.test.js [hunk 3]
```diff
@@ -499,14 +499,37 @@ async function executeCycle(
...
-                      await expect(takaturnDiamondParticipant_1.withdrawFund(termId)).to.emit(
-                          takaturnDiamond,
-                          "OnFundWithdrawn"
+                      await expect(takaturnDiamondParticipant_1.withdrawFund(termId))
+                          .to.emit(takaturnDiamond, "OnFundWithdrawn")
+                          .withArgs(termId, participant_1.address, participant_1.address, 550000000)
+                  })
+
+                  // This happens in the 1st cycle
+                  it("allows the beneficiary to claim the fund to a different address", async function () {
```
Comment (MohammedTaka) [inline test/unit/02-combined-tests/combined.test.js:512]: Please also do a test for withdrawing collateral

**Review by MohammedTaka — CHANGES_REQUESTED** [review]

File: contracts/facets/CollateralFacet.sol (hunk 1 above)
Comment (maikelordaz) [reply contracts/facets/CollateralFacet.sol]: You are right, missed it. Fix it on commit 3a8204f and fix tests on commit 8c15233

File: contracts/facets/YGFacetZaynFi.sol (hunk 2 above)
Comment (maikelordaz) [reply contracts/facets/YGFacetZaynFi.sol]: Fix on commit 2c652ea
Test on commit cbe5fc6

File: test/unit/02-combined-tests/combined.test.js (hunk 3 above)
Comment (maikelordaz) [reply test/unit/02-combined-tests/combined.test.js:512]: It is on https://github.com/TakafulDAO/takaturn-phase-2/blob/cbe5fc64276858d8a6d75dc5cc402de7212ea262/test/unit/01-by-facet/02-collateral-facet.test.js#L219

**Review by MohammedTaka — APPROVED** [review]

**Pull request merged**